### PR TITLE
Generate config docs from source code

### PR DESCRIPTION
**Pull request by SignorMercurio**
This PR works together with apache/pulsar#16627. Please see more details there.

I'm aware that this PR includes doc changes that should only happen in [apache/pulsar](https://github.com/apache/pulsar). These changes are included intentionally to prevent website build failure due to missing doc files, and are also included in apache/pulsar#16627.

To test it locally, run at `$PULSAR_HOME`:

```shell
$ java -cp `cat distribution/server/target/classpath.txt` org.apache.pulsar.utils.CmdGenerateDocumentation -c org.apache.pulsar.broker.ServiceConfiguration > site2/docs/reference-configuration-broker.md

$ cp site2/docs/reference-configuration-broker.md /path/to/pulsar-site/site2/website-next/docs/
```

### CODE REVIEW

**Comment by SignorMercurio** [comment]
@Anonymitaet PTAL

**Comment by SignorMercurio** [comment]
@urfreespace I've reverted doc changes and this PR is ready for review now.

**Review by urfreespace — COMMENTED** [review]
why `reference-configuration.md` modified here @SignorMercurio  /cc @Anonymitaet PTAL

**Comment by SignorMercurio** [comment]
@urfreespace I've reverted the changes in `*.md` in this PR and only two files are changed now.

File: site2/website-next/sidebars.json [hunk 1]
```diff
@@ -324,7 +324,25 @@
...
+            "reference-configuration-broker",
+            "reference-configuration-client",
+            "reference-configuration-log4j",
+            "reference-configuration-log4j-shell",
+            "reference-configuration-standalone",
+            "reference-configuration-websocket",
+            "reference-configuration-pulsar-proxy",
+            "reference-configuration-zookeeper"
+          ]
+        },
```
Comment (urfreespace) [inline site2/website-next/sidebars.json:345]: changes from sidebars.json also need to be submitted in repo `apache/pulsar` but not here. @SignorMercurio /cc @Anonymitaet

**Review by urfreespace — CHANGES_REQUESTED** [review]

**Comment by SignorMercurio** [comment]
@urfreespace PTAL. Only `custom.css` is changed, but I also deleted `brodocs/documents/*.md` as they should be ignored in version control (see `.gitignore`).

**Review by urfreespace — APPROVED** [review]

**Pull request merged**